Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+                ListUsers();
+            }
+            else {
+                if (RetryConfirmation("Bad user details."))
+                {
+                    PromptAddUserToArray(addToEnd, addToStart);
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
+        }
+
+        //Function to get index of where user should be removed
+        static void PromptRemoveUser(bool removeFromEnd = true, bool removeFromStart = false) {
+            Console.Clear();
+            int index = GetIndex(removeFromEnd, removeFromStart);
+            if (index!=-1)
+            {
+                RemoveUser(index);
+                WriteUsersToFile();
+            }
+            ListUsers();
+        }
+
+        //returns or prompts index for array. Returns -1 if there are issues.
+        static int GetIndex(bool endOfArray = true, bool startOfArray = false) {
+            if (endOfArray)
+            {
+                return userArray.Length;
+            }
+            else if (startOfArray)
+            {
+                return 0;
+            }
+            else if (!endOfArray && !startOfArray)
+            {
+                string indexStr = PromptInput("Enter index:");
+                int.TryParse(indexStr, out int index);
+
+                if (index < 0 || index >= userArray.Length)
+                {
+                    return -1;
+                }
+                else {
+                    return index;
+                }
+            }
+            return -1;
+        }
+
+
+        //Remove user from userArray at index.
+        static void RemoveUser(int index) {
+            string[][] newArray = new string[userArray.Length - 1][];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i <index)
+                {
+                    newArray[i] = userArray[i];
+                }else if (i == index)
+                {
+                    newArray[i] = userArray[i + 1];
+                }else if (i > index)
+                {
+                    newArray[i] = userArray[i + 1];
+                }
+            }
+
+            userArray = newArray;
+        }
+
+        //Sorting users
+        static void SortUsers() {
+            Console.Clear();
+
+            bool ascendingOrder = ReturnSortingOrder();
+
+            if (ascendingOrder)
+            {
+                SortUserArrayInAscOrder();
+            }
+            else {
+                SortUserArrayInDesOrder();
+            }
+
+            WriteUsersToFile();
+            ListUsers();
+        }
+
+        // Function to swap values in userArray
+        static void UserArrayValueSwap(int x, int y) {
+            string[] temp;
+            temp = userArray[x];
+            userArray[x] = userArray[y];
+            userArray[y] = temp;
+        }
+
+        //Ascending order sorting function
+        static void SortUserArrayInAscOrder() {
```

Comment:
This is almost identical to `SortUserArrayInDescOrder`.   You could pass a boolean arg in instead.

Reply:
Combined functions to accept bool argument